### PR TITLE
Add iOS Testing Step

### PR DESCRIPTION
**Pull request by 0xTim**
Allows us to run tests on iOS

### CODE REVIEW

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -67,6 +72,11 @@ on:
         required: false
         default: ''
         description: "Additional 'swift build' flags to be applied on when compiling with MUSL."
+      ios_xcodebuild_flag:
```
Comment (gwynne) [inline .github/workflows/run-unit-tests.yml]: I'd rather this be called `action` than `flag.`

File: .github/workflows/run-unit-tests.yml [hunk 2]
```diff
@@ -224,6 +235,19 @@ jobs:
...
           swift test ${PACKAGE_ROOT} ${WITH_TSAN} ${TEST_FILTER} ${EXTRA_FLAGS}
 
+  ios-unit:
+    if: ${{ !(github.event.pull_request.draft || false) && inputs.ios_scheme_name != '' }}
+    runs-on: macos-latest
+    steps:
+      - name: Select appropriate Xcode version
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: 16.0
```
Comment (gwynne) [inline .github/workflows/run-unit-tests.yml]: I'd suggest `latest-stable` here.

Comment (0xTim) [reply .github/workflows/run-unit-tests.yml]: This currently points to Xcode 15 so won't work with Swift 6

Comment (0xTim) [reply .github/workflows/run-unit-tests.yml]: Ok turns out for some utterly bizarre reason GH are going to remove Xcode 16 from the macOS 14 runners, even though they're marked as `macOS-latest` 🤯 

I've manually set the macOS version to 15 which has Xcode 16 as latest stable

File: .github/workflows/run-unit-tests.yml (hunk 1 above)
Comment (0xTim) [reply .github/workflows/run-unit-tests.yml]: Done

File: .github/workflows/run-unit-tests.yml (hunk 2 above)
Comment (gwynne) [reply .github/workflows/run-unit-tests.yml]: Bizarre is definitely the word for it... Well, maybe they'll have the Sequoia runners up sooner than later for once.

**Review by gwynne — APPROVED** [review]

**Pull request merged**